vignettes/customizing_styler.Rmd

```diff
@@ -0,0 +1,268 @@
+  style_text(string_to_format, style = tidyverse_style, strict = FALSE),
+  style_text(string_to_format, strict = FALSE),
+)
+```
+
+Now let's do the whole styling of a string with just this
+one transformer introduced above. We do this by first creating a style guide 
+with the designated wrapper function `create_style_guide()`.
+It takes transformer functions as input and returns them in a named list that 
+meets the formal requirements for styling functions.
+
+```{r}
+space_after_opening_style <- function(are_you_sure) {
+  create_style_guide(space = if (are_you_sure) styler:::remove_space_after_opening_paren)
+}
+style_text("call( 1,1)", style = space_after_opening_style, are_you_sure = FALSE)  
+```
+
+Well, we probably want:
+
+```{r}
+style_text("call( 1,1)", style = space_after_opening_style, are_you_sure = TRUE)  
+```
+
+Note that the return value of your `style` function may not contain `NULL` 
+elements.
+
+I hope you have acquired a basic understanding of how styler transforms code.
+You can provide your own transformer functions and use `create_style_guide()`
+to create customized code styling. If you do so, there are a few more things you
+should be aware of, which are described in the next section.
+
+# Implementation details
+
+For both spaces and line break information in the nested parse table, we use
+four attributes in total: `lag_newlines`, `newlines`, `spaces`, `lag_spaces`.
+`lag_spaces` is created from `spaces` only just before the parse table is
+serialized, so it is not relevant for manipulating the parse table as
+described above. These columns are to some degree redundant, but with just lag
+or lead, we would loose information on the first or the last element 
+respectively, so we need both.
+
+The sequence in which styler applies rules on each level of nesting 
+is given in the list below:
+
+* call `create_filler()` to initialize some variables.
+* modify the line breaks (modifying `lag_newlines` only based on 
+  `token`, `token_before`, `token_after` and `text`).
+* modify the spaces (modifying `spaces` only based on `lag_newlines`, 
+  `newlines`, `multi_line`, `token`, `token_before`, `token_after` and `text`).
+* modify the tokens (based on `newlines` `lag_newlines`, `spaces` `multi_line`, 
+  `token`, `token_before`, `token_after` and `text`).
+* modify the indention by changing `indention_ref_id` (based on `newlines` 
+  `lag_newlines`, `spaces` `multi_line`, `token`, `token_before`, `token_after` 
+  and `text`).
+
+You can also look it up in the function that applies the transformers: 
+`apply_transformers()`:
+```{r}
+styler:::apply_transformers
+```
+
+This means that the order of the styling is clearly defined and it is for 
+example not possible to modify line breaks based on spacing, because spacing
+will be set after line breaks are set. Do not rely on the column `col1`,
+`col2`, `line1` and `line2` in the parse table in any of your function since 
```

Comment:
So not sure what priority that has since it is not totally straight forward (see #147).